pkg/optimization/device_perf_linux.go

```diff
@@ -43,79 +45,96 @@ func (deviceHelper *DeviceHelper) DiskSupportsPerfOptimization(diskPerfProfile,
 
 // OptimizeDiskPerformance optimizes device performance by setting tuning block device settings
 func (deviceHelper *DeviceHelper) OptimizeDiskPerformance(nodeInfo *NodeInfo, devicePath,
-	perfProfile, accountType, diskSizeGibStr, diskIopsStr, diskBwMbpsStr string) (err error) {
-	klog.V(2).Infof("OptimizeDiskPerformance: Tuning settings for %s", devicePath)
+	perfProfile, accountType, diskSizeGibStr, diskIopsStr, diskBwMbpsStr string, deviceSettings map[string]string) (err error) {
 
 	if nodeInfo == nil {
 		return fmt.Errorf("OptimizeDiskPerformance: Node info is not provided. Error: invalid parameter")
 	}
 
-	queueDepth, nrRequests, scheduler, maxSectorsKb, readAheadKb, err := getOptimalDeviceSettings(nodeInfo, DiskSkuMap, perfProfile, accountType, diskSizeGibStr, diskIopsStr, diskBwMbpsStr)
+	deviceName, err := getDeviceName(devicePath)
 	if err != nil {
-		return fmt.Errorf("OptimizeDiskPerformance: Failed to get optimal settings for profile %s accountType %s device %s Error: %v", perfProfile, accountType, devicePath, err)
+		return fmt.Errorf("OptimizeDiskPerformance: Could not get deviceName for %s. Error: %v", devicePath, err)
 	}
 
-	deviceName, err := getDeviceName(devicePath)
+	klog.V(2).Infof("OptimizeDiskPerformance: Tuning settings for devicePath %s deviceName %s perfProfile %s accountType %s",
+		devicePath,
+		deviceName,
+		perfProfile,
+		accountType)
+
+	switch strings.ToLower(perfProfile) {
+	case consts.PerfProfileBasic:
+		return optimizeDiskPerformanceForBasicProfile(nodeInfo,
```

Comment:
Instead of having separate functions for handling the basic vs manual profile, could you simply return a constant `deviceSettings map[string]string` that would have the same effect if passed to the current implementation of `optimizeDiskPerformanceForManualProfile`?

Reply:
Thanks. separated out the get and update parts, so future profiles would just require a new get for them.